cmake/partition_manager.cmake

```diff
@@ -126,16 +126,42 @@ list(APPEND header_files ${ZEPHYR_BINARY_DIR}/${generated_path}/pm_config.h)
 # Add subsys defined pm.yml to the input_files
 list(APPEND input_files ${PM_SUBSYS_PREPROCESSED})
 
+  # Getting RAM info is not trivial. The code below should be replaced with
+  # a simple symbol from the devicetree information. See
+  # https://github.com/zephyrproject-rtos/zephyr/pull/24552
+  # How 'ram_size' is set is also affected by this.
+  set(ram_base 0x20000000)
 if (DEFINED CONFIG_SOC_NRF9160)
   # See nRF9160 Product Specification, chapter "UICR"
   set(otp_start_addr "0xff8108")
   set(otp_size 756) # 189 * 4
+
+    # RAM is divided into separate partitions in DT, hence we can not use the
+    # 'CONFIG_SRAM_SIZE' value here as it only contains a subset of the
+    # actual RAM.
+    set(ram_size 0x40000) # TODO get this from DT when available
 elseif (DEFINED CONFIG_SOC_NRF5340_CPUAPP)
   # See nRF5340 Product Specification, chapter Application Core -> ... "UICR"
   set(otp_start_addr "0xff8100")
   set(otp_size 764)  # 191 * 4
+
+    # RAM is divided into separate partitions in DT, hence we can not use the
+    # 'CONFIG_SRAM_SIZE' value here as it only contains a subset of the
+    # actual RAM.
+    set(ram_size 0x80000) # TODO get this from DT when available
+  else()
+    # TODO get this from DT when available
+    math(EXPR ram_size "${CONFIG_SRAM_SIZE} * 1024" OUTPUT_FORMAT HEXADECIMAL)
```

Comment:
This is called `SRAM_SIZE` in Kconfig and DT.
Any reason why not stick to `sram`, instead of switching to `ram` ?

Reply:
No, I will change it to sram